// @astrojs/markdown-remark/dist/remark-mdxish.js
import { mdxjs } from "micromark-extension-mdxjs";
import { mdxFromMarkdown, mdxToMarkdown } from "./mdast-util-mdxish.js";
function remarkMdxish(options = {}) {
  const data = this.data();
  add("micromarkExtensions", mdxjs(options));
  add("fromMarkdownExtensions", makeFromMarkdownLessStrict(mdxFromMarkdown()));
  add("toMarkdownExtensions", mdxToMarkdown());
  function add(field, value) {
    const list = data[field] ? data[field] : data[field] = [];
    list.push(value);
  }
}
function makeFromMarkdownLessStrict(extensions) {
  extensions.forEach((extension) => {
    ["mdxJsxFlowTag", "mdxJsxTextTag"].forEach((exitHandler) => {
      if (!extension.exit || !extension.exit[exitHandler])
        return;
      extension.exit[exitHandler] = chainHandlers(fixSelfClosing, extension.exit[exitHandler]);
    });
  });
  return extensions;
}
const selfClosingTags = /* @__PURE__ */ new Set([
  "area",
  "base",
  "br",
  "col",
  "embed",
  "hr",
  "img",
  "input",
  "link",
  "meta",
  "source",
  "track",
  "wbr"
]);
function fixSelfClosing() {
  const tag = this.getData("mdxJsxTag");
  if (tag.name && selfClosingTags.has(tag.name))
    tag.selfClosing = true;
}
function chainHandlers(...handlers) {
  return function handlerChain(token) {
    handlers.forEach((handler) => handler.call(this, token));
  };
}
export {
  remarkMdxish as default
};
